docs/functionalabstractions/concretetypes/associative.md

```diff
@@ -3,4 +3,257 @@ id: associative
+
+## New Types
+
+So far we have not described some very basic ways of combining that are associative, such as integer addition.
+
+The reason for this is that some data types support more than one way of combining them that is associative. For example, both integer addition and multiplication are associative.
+
+This creates a potential issue when using the type class pattern because the Scala compiler looks up implicit values based on their type, so if there are two different implicit values of a given type the Scala compiler does not know which one to use.
+
+For example, here is what happens if we try to define `Associative` instances for `Int` for both addition and multiplication.
+
+```scala mdoc:fail
+implicit val IntSumAssociative: Associative[Int] =
+  new Associative[Int] {
+    def combine(left: => Int, right: => Int): Int =
+      left + right
+  }
+
+implicit val IntProdAssociative: Associative[Int] =
+  new Associative[Int] {
+    def combine(left: => Int, right: => Int): Int =
+      left * right
+  }
+
+2 <> 3 // does not compile
+```
+
+This makes sense because there are indeed two different instances of `Associative[Int]` and there is no basis for choosing between them.
+
+We could just define one of these instances as "primary" like some other functional programming libraries do and relegate the other to second class status but that would be arbitrary and reflect a lack of compositionality.
+
+Instead we solve the problem in a way that fits with Scala's implicit resolution mechanism. If the Scala compiler looks up implicit values based on their types, then if we want two values we need two types.
+
+We can easily do this with ZIO Prelude's new type functionality.
+
+New types allow us to define new types that "wrap" existing types in a way that has no overhead at runtime. We can also define these new types so that the Scala compiler actually knows that the new type is a subtype of the underlying type.
+
+Using this technique, we can define new types `Sum` and `Prod` that can be combined using addition and multiplication, respectively. ZIO Prelude provides these and similar new types such as `And` and `Or` for logical conjunction and disjunction in the `zio.prelude.newtypes` package.
+
+We can wrap any existing type in a new type using the `apply` or `wrap` operators on the new type object.
+
+```scala mdoc:reset
+import zio.prelude._
+import zio.prelude.newtypes._
+
+val sumInt: Sum[Int] =
+  Sum(1)
+
+val prodInt: Prod[Int] =
+  Prod.wrap(2)
+```
+
+We can unwrap any new type to get back the original type using the `unwrap` operator on the new type object.
+
+```scala mdoc
+val int: Int =
+  Sum.unwrap(sumInt)
+```
+
+However, we will typically not need to do that for new types like `Sum` and `Prod` because they are subtypes of `Int`.
+
+```scala mdoc:nest
+val int: Int =
+  prodInt
+```
+
+Let's use these types to solve our problem from above.
+
+```scala mdoc
```

Comment:
The fact that the examples that follow here compile without ambiguity makes your point all right. It might be even more obvious if you'd provide the unwrapped resulting `Int` value in the examples.

Reply:
Done.